src/builder/modules/renderers/drawTexture.ts

```diff
@@ -169,87 +183,95 @@ function drawNearestNeighbor(
   options: DrawNearestNeighborOptions
 ): void {
   const rotateOption = options.rotate ?? { kind: "None" };
-  const flipOption = options.flip ?? { kind: "None" };
+  const flipOption = options.flip ?? "None";
   const blendOption = options.blend ?? { kind: "None" };
   const pixelateOption = options.pixelate ?? false;
+  const glintOption = options.glint ?? undefined;
 
   const { canvasWithContext, sx, sy, sw, sh, dx, dy, dw, dh } =
     makeInitialValues(texture, coordinates, pixelateOption);
 
   if (sw > 0 && sh > 0 && dw > 0 && dh > 0) {
     const imageData = canvasWithContext.context.getImageData(sx, sy, sw, sh);
-
     const pix = imageData.data;
-
     const temp = makeCanvasWithContext(dw, dh);
 
     const deltax = dw / sw;
     const deltay = dh / sh;
 
-    const pixwInitial = Math.floor(deltax);
-    const pixhInitial = Math.floor(deltay);
-
-    const pixw = pixwInitial < deltax ? pixwInitial + 1 : pixwInitial;
-    const pixh = pixhInitial < deltay ? pixhInitial + 1 : pixhInitial;
+    const pixw = Math.ceil(deltax);
+    const pixh = Math.ceil(deltay);
 
-    const blend: [number, number, number] | null =
+    const blendColor: Color | null =
       blendOption.kind === "MultiplyHex"
-        ? hexToRGB(blendOption.hex)
-        : blendOption.kind === "MultiplyRGB"
-          ? [blendOption.r, blendOption.g, blendOption.b]
-          : null;
+        ? hexToColor(blendOption.hex)
+        : blendOption.kind === "MultiplyColor"
+        ? blendOption.color
+        : null;
+
+    const replace: [Color[], Color[]] | null =
```

Comment:
What do you think about naming these blends:

* `blendMultiply`
* `blendReplace`

?